frontend/packages/ux-editor/src/hooks/queries/useDataModelMetadataQuery.ts

```diff
@@ -9,6 +9,7 @@ export const useDataModelMetadataQuery = (
   app: string,
   layoutSetName: string,
   dataModelName: string,
+  enabled?: boolean,
```

Comment:
Why is this needed? 🙈

Reply:
I added this because in the `useValidDataModels` hook, there are two fetches: the first for data models and once that is fetched, we proceed to fetch `dataModelMetadata`. The fetching of `dataModelMetadata` depends on the result from the initial fetch of data models.

Reply:
The enabled option is an API feature from TanStack Query, often used with [dependent queries](https://tanstack.com/query/v5/docs/framework/react/guides/dependent-queries).

@lassopicasso, what do you think about refactoring this query to use object parameters instead of positional parameters? This refactor would improve readability where it's used.

Here's the refactored code:
```typescript
  export const useDataModelMetadataQuery = ({
   org,
   app,
   layoutSetName,
   dataModelName
  }: UseDataModelMetadataQuery, options: QueryOptions) => {...rest-of-the-code...}
```
Then it's easier to read from there it used.

```typescript
  const { data } = useDataModelMetadataQuery({ org, app, layoutSetName, dataModelName}, {
     enabled: isDataModelFetched
  });
```

Reply:
Nice tip, changes it to object parameters.

Reply:
Looks more readable indeed!